Metrics: Expose functions to initialize counters at zero

joanlopez opened the pull request:
**What this PR does / why we need it**:

It changes a bit the  `newCounterStartingAtZero` and `newCounterVecStartingAtZero` into public functions in the `metricutil` package for the sake of reusability.

**Special notes for your reviewer**:

I also started using these functions for `SecretsManager` (Envelope Encryption) metrics, as suggested by @hairyhenderson.
In fact, that's why I marked it as `backport v9.0.x`, because we detected that possible improvement while investigating some incidents in `beta-1` directly or indirectly related with encryption operations, so imho it's that kind of minor improvement that we'll appreciate to have in `v9-stable`.

The idea of the refactor came from the need for having similar behavior (counters starting at zero) for `SecretsManager` metrics initialization and also because as we're moving into a more package-oriented architecture (instead of the monolithic one that we have now where all metrics are in the same package, all database operations are together as well, etc), I guessed that kind of function could be useful in the near future.

Thanks!

joanlopez committed:
Metrics: Expose functions to initialize counters at zero

joanlopez committed:
Merge remote-tracking branch 'upstream/main' into metrics/refactor-vec-zero

xlson reviewed (approved):
I think the changes makes a lot of sense. But I am wondering if shouldn't just wait with backporting it until after stable so that it goes into 9.0.1 instead of 9.0.0-beta3. What would the downside of waiting be? Does it hold back other changes in enterprise?

joanlopez commented:
> I think the changes makes a lot of sense. But I am wondering if shouldn't just wait with backporting it until after stable so that it goes into 9.0.1 instead of 9.0.0-beta3. What would the downside of waiting be? Does it hold back other changes in enterprise?

Agree! So, I've:
- Removed the backport label.
- Changed the milestone to `v9.0.1`.

So, I think we can merge it and add the backport label later (or create the backport manually if actually needed and/or in case the automated process fails).

WDYT?
